scarb/src/compiler/plugin/proc_macro/host.rs

```diff
@@ -34,6 +34,7 @@ use tracing::{debug, trace_span};
 
 const FULL_PATH_MARKER_KEY: &str = "macro::full_path_marker";
 const DERIVE_ATTR: &str = "derive";
+const USED_ATTR_MARKER: &str = "proc_macro::";
```

Comment:
If we rewrite a function, but leave the original tag on it - it will be expanded recursively.

Reply:
Changing the tag is exactly against what @orizi said about the performance implications of rewriting too many things 😕 I'm not sure right now what to do here.

Reply:
Except, it's what we do when you need to rewrite the node *anyway*. If the node is not modified, then there is no need to change the tag. But since each execution of proc macro attribute does rewrite the node, we cannot return the original tag on a generated function - as it would cause the plugin to be called recursively.